be/src/exec/pipeline/hashjoin/hash_join_probe_operator.cpp

```diff
@@ -4,50 +4,78 @@
 
 namespace starrocks {
 namespace pipeline {
+
 HashJoinProbeOperator::HashJoinProbeOperator(OperatorFactory* factory, int32_t id, const string& name,
-                                             int32_t plan_node_id, HashJoinerPtr hash_joiner)
-        : OperatorWithDependency(factory, id, name, plan_node_id), _hash_joiner(hash_joiner) {
-    _hash_joiner->ref();
+                                             int32_t plan_node_id, HashJoinerPtr join_prober,
+                                             HashJoinerPtr join_builder)
+        : OperatorWithDependency(factory, id, name, plan_node_id),
+          _join_prober(std::move(join_prober)),
+          _join_builder(std::move(join_builder)) {}
+
+Status HashJoinProbeOperator::close(RuntimeState* state) {
+    RETURN_IF_ERROR(_join_prober->unref(state));
```

Comment:
What if error? Will `_join_builder` leaked?

Reply:
No, all the classes are exception safe until now. It will release the resource in the destructor, if `unref` returns error.
And `close()` methods of all the operators and contexts in pipeline and non-pipeline always return `Status::OK`, it just a compatibility code.

Reply:
But it is a little confusing. Maybe I could remove `RETURN_IF_ERROR` in all the `close()` methods In the next PR.